### PR TITLE
fix(nuxt3-ssr): layout brakes with incompressible components

### PR DESCRIPTION
**Pull request by MaxPostema**
Components that are too big to fit in the layout should not destroy it.
- [x] Fix the layout braking
- [x] Add a component to create a nice horizontal scrollbar
- [x] Add the scroll button from the design

### CODE REVIEW

**Review by mswertz — APPROVED** [review]
great. Unrelated issue is the fact that 'no data' cohorts are still in the query

File: apps/nuxt3-ssr/components/HorizontalScrollHelper.vue [hunk 1]
```diff
@@ -0,0 +1,61 @@
+<script setup>
```
Comment (connoratrug) [inline apps/nuxt3-ssr/components/HorizontalScrollHelper.vue:1]: all of the 'new' components use typescript,

File: apps/nuxt3-ssr/components/HorizontalScrollHelper.vue [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  scrollAtStart.value = event.target.scrollLeft === 0;
+  scrollAtEnd.value =
+    event.target.scrollLeft + event.target.offsetWidth ===
+    event.target.scrollWidth;
+}
+
+function smoothScroll(value) {
+  scrollElement.value.scrollTo({
+    left: scrollElement.value.scrollLeft - value,
+    behavior: "smooth",
```
Comment (connoratrug) [inline apps/nuxt3-ssr/components/HorizontalScrollHelper.vue:27]: nice :) , did you take a look at: https://tailwindcss.com/docs/scroll-snap-type ?

**Review by connoratrug — APPROVED** [review]

File: apps/nuxt3-ssr/components/HorizontalScrollHelper.vue (hunk 1 above)
Comment (connoratrug) [reply apps/nuxt3-ssr/components/HorizontalScrollHelper.vue]: i added typing

**Pull request merged**